src/utilities/element-utilities-extender-factory.js

```diff
@@ -1050,45 +1065,68 @@ module.exports = function () {
         }
-        else {
+        else if(templateType === 'dissociation'){
           newEdge = elementUtilities.addEdge(process.id(), newNode.id(), 'production');
+        }else{
+          //Group right or top elements in group id 1
+          newEdge = elementUtilities.addEdge(process.id(), newNode.id(), 'production', undefined, undefined, 1);
         }
 
         newEdge.data('justAdded', true);
 
         //update the y position
         yPosition += macromoleculeHeight + tilingPaddingVertical;
       }
+      if(templateType === 'association' || templateType == 'dissociation'){
+        //Create the complex including macromolecules inside of it
+        //Temprorarily add it to the process position we will move it according to the last size of it
+        var complex = elementUtilities.addNode(processPosition.x, processPosition.y, 'complex');
+        complex.data('justAdded', true);
+        complex.data('justAddedLayoutNode', true);
+
+        //If a name is specified for the complex set its label accordingly
+        if (complexName) {
+          complex.data('label', complexName);
+        }
 
-      //Create the complex including macromolecules inside of it
-      //Temprorarily add it to the process position we will move it according to the last size of it
-      var complex = elementUtilities.addNode(processPosition.x, processPosition.y, 'complex');
-      complex.data('justAdded', true);
-      complex.data('justAddedLayoutNode', true);
-
-      //If a name is specified for the complex set its label accordingly
-      if (complexName) {
-        complex.data('label', complexName);
-      }
-
-      //create the edge connnected to the complex
-      var edgeOfComplex;
-      if (templateType === 'association') {
-        edgeOfComplex = elementUtilities.addEdge(process.id(), complex.id(), 'production');
-      }
-      else {
-        edgeOfComplex = elementUtilities.addEdge(complex.id(), process.id(), 'consumption');
+        //create the edge connnected to the complex
+        var edgeOfComplex;
+        if (templateType === 'association') {
+          edgeOfComplex = elementUtilities.addEdge(process.id(), complex.id(), 'production');
+        }
+        else {
+          edgeOfComplex = elementUtilities.addEdge(complex.id(), process.id(), 'consumption');
+        }
+        edgeOfComplex.data('justAdded', true);
+
+        //Create the macromolecules inside the complex
+        for (var i = 0; i < numOfMacromolecules; i++) {
+          // Add a macromolecule not having a previously defined id and having the complex created in this reaction as parent
+          var newNode = elementUtilities.addNode(complex.position('x'), complex.position('y'), "macromolecule", undefined, complex.id());
+          newNode.data('justAdded', true);
+          newNode.data('label', macromoleculeList[i]);
+          newNode.data('justAddedLayoutNode', true);
+        }
+      }else{
```

Comment:
Let's follow the format of
`if ( ) {

}
else {

}`
here and a few more places since it is generally the case.